### ISSUE TITLE
Error span causes refactoring suggestions

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/vscode#34481]
- VSCode Version: Code - Insiders 1.17.0-insider (b0c0634339c6f64a5c7f08b56fe732fb9154d3c4, 2017-09-15T05:24:23.107Z)
- OS Version: Windows_NT x64 10.0.15063
- Extensions: none
**TypeScript Version:**  nightly (2.6.0-dev.20170915)
---

Moving from Microsoft/TypeScript#18507 as this looks like a vscode issue.

**Code**

```ts
function f(): number {
    return {
        x: 1,
    }
}
```

The entire `return` statement has a red squiggly line under it.
Click anywhere in the error span and there will be a suggestion to refactor to an inner function.
Remove `: number` and there are no refactoring suggestions.
(Tested in vscode.)

**Expected behavior:**

Presence of a compile error doesn't cause a refactoring to be suggested.
Especially because it looks exactly like a code fix, but extracting to an inner function won't fix the compile error.

**Actual behavior:**

It does.
In fact, hovering over the light bulb shows "Show Fixes", but this is a refactoring, not a fix.

**Comment by mjbvz** [comment]
Pushing this out as it depends on #34664
